frontend/src/console/components/Hero/Hero.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from 'react';
+import './styles.scss';
+import {
+  ConsolePage,
+  ConsolePageCustomizationContainer
+} from 'console/components';
+import { Col, Row } from 'react-bootstrap';
+import { InstancesModel } from 'console/models';
+import {
+  CollapseEditArea,
+  ColorInputField,
+  HeroPreview,
+  ImageUploadField,
+  TextInputField
+} from 'ui/components';
+import { connect } from 'react-redux';
+import { RootState } from 'global/state';
+import { WrappedMessage } from 'utils/intl';
+import {
+  clearErrorMessage,
+  updateHeroText,
+  updateImages,
+  updateStaticContentOverridesFieldValue,
+  updateThemeFieldValue
+} from 'console/actions';
+import { StaticContentOverrides } from 'ocim-client';
+import messages from './displayMessages';
+
+interface State {
+  [key: string]: string;
+  title: string;
+  subtitle: string;
+}
+
+interface ActionProps {
+  clearErrorMessage: Function;
+  updateHeroText: Function;
+  updateThemeFieldValue: Function;
+  updateImages: Function;
+  updateStaticContentOverridesFieldValue: Function;
+}
+interface StateProps extends InstancesModel {}
+interface Props extends StateProps, ActionProps {}
+
+export class HeroComponent extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      title: '',
+      subtitle: ''
+    };
+    if (this.doesHomepageOverlayHtmlExist(this.props)) {
+      const homepageOverlayHtml = props.activeInstance.data!
+        .draftStaticContentOverrides!.homepageOverlayHtml!;
+      const heroTextRegex = /<h1>(.*)<\/h1><p>(.*)<\/p>/;
+      const matched = heroTextRegex.exec(homepageOverlayHtml as string);
+      if (matched) {
+        this.state = {
+          title: matched[1],
+          subtitle: matched[2]
+        };
+      }
+    }
+  }
+
+  private doesHomepageOverlayHtmlExist = (props: Props) => {
```

Comment:
Since this function is bound to the class, you can use `this` inside, so there's no need to pass the props.